docs/docsite/rst/user_guide/complex_data_manipulation.rst

```diff
@@ -0,0 +1,243 @@
+.. _complex_data_manipulation:
+
+Complex data manipulation
+#########################
+
+In many cases, you need to do some complex operation with your variables, while Ansible is not recommended as a data processing/manipulation tool, you can use the existing Jinja2 templating in conjunction with the many added Ansible filters, lookups and tests to do some very complex transformations.
+
+Let's start with a quick definition of each type of plugin:
+  - lookups: Mainly used to query 'external data', in Ansible these were the primary part of loops using the ``with_<lookup>`` construct, but they can be used independently to return data for processing. They normally return a list due to their primary function in loops as mentioned previously. Used with the ``lookup`` or ``query`` Jinja2 operators.
+  - filters: used to change/transform data, used with the ``|`` Jinja2 operator.
+  - tests: used to validate data, used with the ``is`` Jinja2 operator.
+
+.. _note:
+   * Some tests and filters are provided directly by Jinja2, so their availability depends on the Jinja2 version, not Ansible.
+
+.. _for_loops_or_list_comprehensions:
+
+Loops and list comprehensions
+=============================
+
+Most programmers are used to use loops (``for``, ``while``, etc.) and list comprehensions to do transformations on lists including lists of objects, Jinja2 has a few filters that provide this functionality: ``map``, ``select``, ``reject``, ``selectattr``, ``rejectattr``.
```

Comment:
Are you sure the audience are only programmers ? I like to think the power of Ansible is to widen the audience than sysadmin as well, I'm a sysadmin and while I can write bash or python script (I know list comprehension but not sure every people reading are aware of what it is) I would not call me programmer.

If this first part is to do an analogy with "standard" programming language, we could write that differently, like ``programming languages have loops .... lists of objects``

Reply:
Done.